polars/polars-ops/src/series/ops/arg_min_max.rs

```diff
@@ -222,32 +217,108 @@ where
     }
 }
 
-fn arg_min_slice<T>(vals: &[T], is_sorted: IsSorted) -> Option<usize>
+fn arg_min_numeric<'a, T>(ca: &'a ChunkedArray<T>) -> Option<usize>
+where
+    T: PolarsNumericType,
+    for<'b> &'b [T::Native]: ArgMinMax,
+{
+    match ca.is_sorted_flag2() {
+        IsSorted::Ascending => Some(0),
+        IsSorted::Descending => Some(ca.len() - 1),
+        IsSorted::Not => {
+            ca.downcast_iter()
+                .fold((None, None, 0), |acc, arr| {
+                    if arr.len() == 0 {
+                        // argminmax assumes not empty
+                        return acc;
+                    }
+                    let chunk_min_idx: usize = arr.values().as_slice().argmin();
+                    let chunk_min_val: T::Native = arr.value(chunk_min_idx);
+                    match acc {
+                        (None, None, acc_offset) => (
+                            Some(chunk_min_idx + acc_offset),
+                            Some(chunk_min_val),
+                            acc_offset + arr.len(),
+                        ),
+                        (Some(acc_min_idx), Some(acc_min_val), acc_offset) => {
+                            if chunk_min_val < acc_min_val {
+                                (
+                                    Some(chunk_min_idx + acc_offset),
+                                    Some(chunk_min_val),
+                                    acc_offset + arr.len(),
+                                )
+                            } else {
+                                (Some(acc_min_idx), Some(acc_min_val), acc_offset + arr.len())
+                            }
+                        }
+                        _ => unreachable!(),
+                    }
+                })
+                .0
+        }
+    }
+}
+
+fn arg_max_numeric<'a, T>(ca: &'a ChunkedArray<T>) -> Option<usize>
+where
+    T: PolarsNumericType,
+    for<'b> &'b [T::Native]: ArgMinMax,
+{
+    match ca.is_sorted_flag2() {
+        IsSorted::Ascending => Some(ca.len() - 1),
+        IsSorted::Descending => Some(0),
+        IsSorted::Not => {
+            ca.downcast_iter()
+                .fold((None, None, 0), |acc, arr| {
+                    if arr.len() == 0 {
+                        // argminmax assumes not empty
+                        return acc;
+                    }
+                    let chunk_max_idx: usize = arr.values().as_slice().argmax();
```

Comment:
This seems to ignore the null values that are set in the validity bits. The values that are masked out can have any arbitrary value, so we are not allowed to assume anything here.

I think that if `null_count == 0`, we can continue with this fast argmax branch, but if not we should take the old branch to ensure correctness.

Reply:
I'm not entirely sure about this, but I believe that the current polars code also doesn't consider the validity bits (at least in the case of a single chunk) - see code below.

However, I'm certainly open to add this functionality to argminmax https://github.com/jvdd/argminmax/issues/14#issuecomment-1495658702
Just wondering about what the desired / default behavior would be; ignoring the NULL indices?

---

Code :arrow_down: 
```rs
    let s = Series::new("a", vec![Some(1), None, Some(3)]);
    let arr = s.i32().unwrap().chunks()[0].as_ref();
    println!("{:?}", arr);
    println!("{:?}", arr.validity().unwrap());
    println!("argmin: {:?}", s.arg_min().unwrap()); // returns 1 - which is None...
    println!("argmax: {:?}", s.arg_max().unwrap());
```

Output :arrow_down: 
```
Int32[1, None, 3]
[0b_____101]
argmin: 1
argmax: 2
```

Reply:
That's because `None < Some(value)` in our ordering. I think for now we should only add this for cases where all values are valid. Adding an arrow dependency via `jvdd/argminmax` will definitely run out of sync meaning we compile arrow twice, which is a huge dependency.

Let's start with the easy case first and then we can figure out the more complicate (nulls) in a later time. :)

Reply:
You are completely right. Did not realize there was an ordering for optional values. I'll update this PR accordingly!

Furthermore - I do think it should be possible to integrate this into argminmax, without necessarily depending on `arrow2`! If you can pass smth like an `Option<BitMask>` to the argminmax functions, you can achieve this functionality without having to depend on `arrow2` :thinking: